ui/src/components/Secrets.vue

```diff
@@ -517,6 +581,58 @@ function removeSecret(counter) {
     secrets.value[0] = { key: "", value: "", file: [] }
   }
 }
+
+function generateSecureRandomString(length, includeUppercase, includeLowercase, includeNumbers, includeSpecial) {
+  const chars = [];
+  const result = [];
+  if (includeUppercase) {
```

Comment:
While we do make sure, that after this we include at least one upper case char in the result. We also weaken the password, as an attacker would now, that the first char would likely be an upper case char.

A better approach would be to use the for loop inside an while loop and test each generated candidate, if it includes all needed char classes.

So pseudo code:
```javascript
const chars = []
const predicates = []
if (includeUppercase) {
  chars.push(...'ABC..YZ')
  predicates.push((candidate) => candidate.search('[ABC..YZ]')
}
...
if (!predicates) { setErrorMessage("At least one ...") }
while (true) {
  const result = []
  for (let i=0; i < length; i++) {
    result.push(getRandomCharacter(chars.join(''))
  }
  const candidate = result.join('')
  if (predicates.every((p) => p(candidate) {
    return candidate
  }
}
```

We should probably add a counter to the while loop and alert, if we are doing too many loops without a valid candidate.

Reply:
If the classes are configured by an hash, we could use something like the following as a starting point to make the classes completely controllable by the specific backend.

```javascript
function generateSecureRandomString(length, includeUppercase, includeLowercase, includeNumbers, includeSpecial) {
  const classes = {
    upperCase: {chars: "ABCDEFGHIJKLMNOPQRSTUVWXYZ", enabled: includeUppercase},
    lowerCase: {chars: "abcdefghijklmnopqrstuvwxyz", enabled: includeLowercase},
    digits: {chars: "0123456789", enabled: includeNumbers},
    special: {chars: "$%!,.+-_/*()[]{}", enabled: includeSpecial}
  }
  const allClassesInCandidate = (candidate) => Object.values(classes)
    .filter(v => v.enabled)
    .map(v =>
      candidate => candidate.split('').some((a) => v.chars.indexOf(a) >= 0))
    .every(predicate => predicate(candidate))
  const chars = Object.values(classes)
    .filter(v => v.enabled)
    .map(v => v.chars)
    .join('')

  if (chars.length === 0) {
    setErrorMessage('At least one character type must be selected to create a random string.');
    return '';
  }
  for (let counter = 0; counter < 20; counter++) {
    let result = ''
    for (let i = 0; i < length; i++) {
      result += getRandomCharacter(chars);
    }
    if (allClassesInCandidate(result)) {
      return result
    }
  }
  setErrorMessage('No good password has been found after many rounds of trying')
  return '';
}
```

This is compatible with the current implementation, but could be changed to take `classes` as an argument instead of the four feature flag parameters.

The `classes` Object could be extended with a description and be used to generate the menu. The `enabled` flag could be inserted/controlled by the menu.